_gsocproposals/2022/proposal_Root-interactive-debugging-rdf.md

```diff
@@ -0,0 +1,39 @@
+---
+title: ROOT - An interactive debugging experience for RDataFrame
+layout: gsoc_proposal
+project: ROOT
+year: 2022
+difficulty: medium
+duration: 350
+mentor_avail: June-August
+organization:
+  - CERN
+---
+
+# Description
+
+[ROOT](https://root.cern) provides a high-level interface for data analysis and manipulation, [RDataFrame](https://root.cern/doc/master/classROOT_1_1RDataFrame.html). It lets users perform complex operations on data with few characters typed, in C++ and Python. Thanks to its simple API, RDataFrame has seen wide adoption from HEP analysts, including for production use. At the same time, it still offers few facilities useful to inspect intermediate results of computations and debug logical errors in user-defined expressions.
+
+## Task ideas and expected results
+
+In this project the student will add features to RDataFrame useful to inspect the state of the data at selected points of the data processing loop, enabling users to verify and inspect logical errors in their analysis code in a much more ergonomic fashion than what is currently available. The student will gain practical experience in the software development cycle of a real-world product and with the tools and best practices employed therein.
+
+## Evaluation Task
+
+Interested students please contact Enrico (see **Mentors** below) for a warm-up and evaluation task.
+
+## Requirements
+
+  * C++
```

Comment:
that's a bit too unspecific. How much experience? any fields of application, etc?

Reply:
i qualified it a bit more